### DISCUSSION TITLE
Panic on migration, 1.27.0

### DISCUSSION THREAD

**Discussion by lretq** [dani-garcia/vaultwarden#3089] [Q&A] [answered]
### Subject of the issue

Since upgrading to the 1.27.0 version, with the groups feature support, vaultwarden panics

### Deployment environment

* vaultwarden version: 1.27.0

* Install method: docker image

* Clients used: not applicable

* Reverse proxy and version: not applicable

* MySQL/MariaDB or PostgreSQL version: Galera cluster behind a loadbalancer, running mariadb version 15.1

* Other relevant details: none

### Steps to reproduce

Start vaultwarden with the loadbalancer configured as DB, it starts and panics.

### Expected behaviour

It shouldn't panic.

### Actual behaviour

Panic

### Troubleshooting data

```
[2023-01-01 17:36:47.220][panic][ERROR] thread 'main' panicked at 'Error running migrations: QueryError(DieselMigrationName { name: "2022-07-27-110000_add_group_support", version: MigrationVersion("20220727110000") }, DatabaseError(Unknown, "Can't create table `vaultwarden`.`groups` (errno: 150 \"Foreign key constraint is incorrectly formed\")"))': src/db/mod.rs:474
   0: vaultwarden::init_logging::{{closure}}
   1: std::panicking::rust_panic_with_hook
   2: std::panicking::begin_panic_handler::{{closure}}
   3: std::sys_common::backtrace::__rust_end_short_backtrace
   4: rust_begin_unwind
   5: core::panicking::panic_fmt
   6: core::result::unwrap_failed
   7: vaultwarden::db::mysql_migrations::run_migrations
   8: vaultwarden::db::DbPool::from_config
   9: vaultwarden::main::{{closure}}
  10: tokio::runtime::park::CachedParkThread::block_on
  11: tokio::runtime::scheduler::multi_thread::MultiThread::block_on
  12: tokio::runtime::runtime::Runtime::block_on
  13: rocket::async_main
  14: vaultwarden::main
  15: std::sys_common::backtrace::__rust_begin_short_backtrace
  16: std::rt::lang_start::{{closure}}
  17: std::rt::lang_start_internal
  18: main
  19: __libc_start_main
  20: _start
```

**Comment by BlackDex** [comment, marked as answer]
See https://github.com/dani-garcia/vaultwarden/wiki/Using-the-MariaDB-%28MySQL%29-Backend#foreign-key-errors-collation-and-charset

**Reply by lretq** [reply]
Thanks, this worked!
